zerver/views/video_calls.py

```diff
@@ -185,6 +186,57 @@ def make_user_authenticated_zoom_video_call(
     return json_success(request, data={"url": res.json()["join_url"]})
 
 
+def get_zoom_server_to_server_access_token() -> str:
+    if settings.VIDEO_ZOOM_CLIENT_ID is None:
+        raise JsonableError(_("Zoom credentials have not been configured"))
+
+    account_id = settings.VIDEO_ZOOM_ACCOUNT_ID
+    client_id = settings.VIDEO_ZOOM_CLIENT_ID.encode("utf-8")
+    client_secret = str(settings.VIDEO_ZOOM_CLIENT_SECRET).encode("utf-8")
+
+    url = "https://zoom.us/oauth/token"
+    data = {"grant_type": "account_credentials", "account_id": account_id}
+
+    client_information = client_id + b":" + client_secret
+    encoded_client = b64encode(client_information).decode("ascii")
+    headers = {"Host": "zoom.us", "Authorization": f"Basic {encoded_client}"}
+
+    response = VideoCallSession().post(url, data, headers=headers)
+    if not response.ok:
+        # {reason: 'Bad request', error: 'invalid_request'} for invalid account ID
+        # {'reason': 'Invalid client_id or client_secret', 'error': 'invalid_client'}
+        raise JsonableError(_("Invalid Zoom credentials"))
+    return response.json()["access_token"]
+
+
+def get_zoom_server_to_server_call(
+    user: UserProfile, access_token: str, payload: ZoomPayload
+) -> str:
+    email = user.delivery_email
+    url = f"https://api.zoom.us/v2/users/{email}/meetings"
+    headers = {"Authorization": f"Bearer {access_token}", "Content-Type": "application/json"}
+    response = VideoCallSession().post(url, json=payload, headers=headers)
+    if not response.ok:
+        if response.json()["code"] == 124:
+            # {code: 124, message: "Invalid access token"}
+            # {code: 124, message: "Access token is expired"}
+            raise InvalidZoomTokenError
+        # {code: 1001, message: "User does not exist: {email}"}
+        raise JsonableError(_("Failed to create Zoom call"))
```

Comment:
For these errors, we'd want the web-app to generate a compose box banner error message, since the user would be unable to use that feature.

Reply:
Yeah, that sounds right. Probably the compose box will want to just plumb through the exception returned by the server, since the error message will likely help indicate what's wrong with their configuration.

Reply:
Added a compose box banner for the case where the user's delivery email does not match a user on the configured Zoom account.

Other errors follow the pattern of the existing Zoom User OAuth integration and show the "Failed to create video call." message via `ui_report.generic_embed_error`.